File: Add Digits.js
/**
 * @param {number} num
 * @return {number}
 */
var addDigits = function(num) {
    // Keep looping until we have a single digit
    while (num >= 10) {
        let total = 0;
        let numStr = num.toString();
        for (let i = 0; i < numStr.length; i++) {
            total += parseInt(numStr[i]);
        }
        num = total;
    }
    return num;
};

File: Add Digits.py
class Solution:
    def addDigits(self, num: int) -> int:
        #keep looping until we have a single digit
        while num >= 10:
            total = 0
            #sum all digits of current number
            for digit in str(num):
                total += int(digit)
            num = total
        return num
